mpt_extension_sdk/pipeline/step.py

```
@@ -0,0 +1,48 @@
+        """Execute the full step lifecycle.
+
+        `post()` runs after `process()` both on success and on failure so the
+        step can perform cleanup or compensating actions. If both `process()`
+        and `post()` fail, the `post()` exception is raised and chained from
+        the original processing error.
+        """
+        await self.pre(ctx)
+        process_error: Exception | None = None
+        try:
+            await self.process(ctx)
+        except Exception as error:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)